### ISSUE TITLE
Difficulty reproducing rMD17 force-MAE benchmarks with new YAML configs

### ISSUE THREAD

**Issue by tirtha-v** [mir-group/nequip-input-files#2]
After the recent major updates to **NequIP** repository, we converted the previous JSON configuration files for the rMD17 dataset to the new YAML format compatible with latest NequIP release (no hyper-parameter changes). However, our validation errors are far above the force-MAE values reported in the NequIP paper [1].

| Molecule    | `l_max` | Reported test MAE (meV / Å) | Our best **validation** MAE (meV / Å) | Epoch |
|-------------|:-------:|------------------------------|----------------------------------------|:-----:|
| Aspirin     |   3     | 8.2                          | ~50                                    | 541   |
| Azobenzene  |   3     | 2.9                          | ~30                                    | 541   |

*see attached YAMLs* for Aspirin and Azobenzene (converted from the config files from this repository with identical hyper-parameters). Attached in .txt format as uploading yaml file isn't supported.

### What we tried
1. **Kept all hyper-parameters identical** to the ones used in the paper.  
2. Verified the **training/validation split** matches the original rMD17 splits.  
3. Confirmed **data normalization** and **unit conversions** follow the defaults in NequIP 0.7.x.  
4. Ran for **> 540 epochs** to monitor convergence.  

### Questions
1. Approximately **how many epochs** did your rMD17 models typically need to converge on these molecules?  
2. Were there any **additional data-processing steps**, training flags, or learning-rate schedules used that are **not captured** in the attached configs?  
3. Are there any **known breaking changes** in NequIP 0.7.x that could explain the larger errors when using the same hyper-parameters?  

Any guidance would be greatly appreciated so we can align our experiments with the methodology in [1].

[aspirin_config.txt](https://github.com/user-attachments/files/21044282/aspirin_config.txt)
[azobenzene_config.txt](https://github.com/user-attachments/files/21044281/azobenzene_config.txt)

Thanks for your time!

[1] Batzner, S., Musaelian, A., Sun, L. et al. E(3)-equivariant graph neural networks for data-efficient and accurate interatomic potentials. Nat Commun 13, 2453 (2022). https://doi.org/10.1038/s41467-022-29939-5

**Comment by Linux-cpp-lisp** [comment]
Hi @tirtha-v ,

From a quick glance one critical difference is in the scaling a shifts. In the old nequip, there was both a per-atom and a global scale and shift:
https://github.com/mir-group/nequip-input-files/blob/main/revmd17/revmd17-aspirin-n950-l1.yaml#L130-L144

While the reference configs used trivial shifts of zero and scales of one *per-atom*, they use dataset derived global shifts and scales. This is very important for training.  In the new code, we combine these scalings into the per-atom scalings:
https://github.com/mir-group/nequip/blob/main/configs/tutorial.yaml#L258-L259

So in your new configs you need to set the per-atom scales and shifts to be dataset derived. To get the same behavior as the old configs, use:
```yaml
per_type_energy_scales: ${training_data_stats:forces_rms}
per_type_energy_shifts: ${training_data_stats:per_atom_energy_mean}
```

(See also: https://nequip.readthedocs.io/en/latest/api/data_stats.html)

Also note that ZBL was not used in the original trainings, although its inclusion is unlikely to be a problem.

**Comment by tirtha-v** [comment]
Thanks, I have reached the benchmark values.

**Comment by Linux-cpp-lisp** [comment]
Great, glad to hear!
